client/src/pages/ProfileListings/hooks/useSitterProfiles.ts

```diff
@@ -0,0 +1,93 @@
+import type { RequestError, ValidationError, OnError } from './../../../interface/ApiData';
+import type { GetSitterProfileParams, SitterProfile } from '../../../interface/Profile';
+import { useEffect, useState, useCallback, useRef } from 'react';
+import { v4 as uuid } from 'uuid';
+import { getSitterProfiles } from '../../../helpers/APICalls/profile';
+import wait from '../../../helpers/wait';
+
+interface Options extends Omit<GetSitterProfileParams, 'page'> {
+  delay?: number;
+  loadOnMount?: boolean;
+}
+
+interface StateItem {
+  id: string;
+  loading: boolean;
+  sitter: SitterProfile | null;
+}
+
+type OnSuccess<R> = (profiles: StateItem[]) => R;
+type MatchProfiles = <R>(onError: OnError<R>, onSuccess: OnSuccess<R>) => R;
+
+const useSitterProfiles = (
+  { delay = 500, loadOnMount = true, limit = 6, ...initialParams }: Options = { delay: 50000, limit: 6 },
+) => {
+  const [error, setError] = useState<ValidationError[] | RequestError | null>(null);
+  const [profiles, setProfiles] = useState<StateItem[]>([]);
+  const isLoading = useRef<boolean>(loadOnMount);
+  const firstLoad = useRef<boolean>(true);
+  const page = useRef<number>(1);
+
+  const setLoadingProfiles = useCallback(
+    (value: boolean, data: SitterProfile[] | null = null) => {
+      if (value) {
+        isLoading.current = value;
+        setProfiles((profiles) => [
+          ...profiles,
+          ...Array(limit)
+            .fill(0)
+            .map(() => ({ id: uuid(), loading: true, sitter: null })),
```

Comment:
Any reason why we are using `uuid`s as the id here?

Reply:
Absolutely, I use a uuid  as a key so react can keep track of the items, as they are changing. If I used an index instead of a uuid, there would be issues with react keeping track of the rendered loading cards. I take this approach so that the transition from the loading profile card to the populated profile card is as smooth as can be, because we will not always have the profile's id to use as a key.